### PR TITLE
Lua v5.4

### PR DESCRIPTION
**Pull request by TraceyC77**
**Summary**

Update lua and packages that rely on it

**Test Plan**

- Built lua, installed successfully
- Entred lua console and printed a message successfully
- Ran lua test script successfully
- Verified vim can edit a file
- Brief testing of other rebuilt software

**Checklist**

- [ ] Package was built and tested against unstable

Resolves getsolus/packages/issues/573
Resolves getsolus/packages/issues/1533

There are 2 commits for xournalapp, the work accidentally got split between them. I would rather merge this as is than mess with the commit history more.

### CODE REVIEW

File: packages/m/mame/package.yml [hunk 1]
```diff
@@ -29,19 +29,18 @@ builddeps  :
     - pkgconfig(libjpeg)
     - pkgconfig(libpulse)
     - pkgconfig(libutf8proc)
-    - pkgconfig(lua)
+    - pkgconfig(lua-5.4)
```
Comment (ReillyBrogan) [inline packages/m/mame/package.yml:32]: If you're not building with lua you don't need the dep

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]

File: packages/m/mame/package.yml [hunk 2]
```diff
@@ -29,19 +29,18 @@ builddeps  :
...
     - pkgconfig(xinerama)
-setup      : |
-    %patch -p1 -i $pkgfiles/Fix-building-with-gcc-13.patch
 build      : |
     # -g2 flag is removed because it's bloating mame64 executable file.
-    # Every 64-bit processor supports SSE2, so I'm enabling it.
-    # Using bundled utf8proc library because it's not in the Solus repository.
+    #
+    # Using bundled Lua since mame expects lua to be compiled with a C++ compiler,
+    # but we're compiling it with a *C* compiler.
```
[the remark this answers is not in the document]
Comment (TraceyC77) [reply packages/m/mame/package.yml:43]: That wasn't written by me, that was @livingsilver94. I see it looks like I wrote it in git blame.

File: packages/m/mame/package.yml (hunk 1 above)
Comment (TraceyC77) [reply packages/m/mame/package.yml:32]: Fixed

File: packages/l/libgaminggear/pspec_x86_64.xml [hunk 3]
```diff
@@ -28,6 +28,7 @@
             <Path fileType="library">/usr/lib64/libgaminggearfx.so.0.15.1</Path>
             <Path fileType="library">/usr/lib64/libgaminggearwidget.so.0</Path>
             <Path fileType="library">/usr/lib64/libgaminggearwidget.so.0.15.1</Path>
+            <Path fileType="data">/usr/share/libgaminggear/cmake/Modules/FindGAMINGGEAR0.cmake</Path>
```
[the remark this answers is not in the document]
Comment (TraceyC77) [reply packages/l/libgaminggear/pspec_x86_64.xml:31]: The cmake flag needs to stay this way or the package doesn't build.
I'll add the cmake folder to the devel patterns.

I've added this to the patterns, but that path still appears in the pspec file?
`- /usr/share/libgaminggear/cmake`

**Review by EbonJaeger — APPROVED** [review]
LGTM, thanks!

**Pull request merged**